wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/CommandScheduler.java

```diff
@@ -271,13 +272,7 @@ public void run() {
       Command command = iterator.next();
 
       if (!command.runsWhenDisabled() && RobotState.isDisabled()) {
-        command.end(true);
-        for (Consumer<Command> action : m_interruptActions) {
-          action.accept(command);
-        }
-        m_requirements.keySet().removeAll(command.getRequirements());
-        iterator.remove();
-        m_watchdog.addEpoch(command.getName() + ".end(true)");
+        cancel(command);
```

Comment:
Just a note that using `cancel` here means that we put this on the pending queue because `m_inRunLoop` is true here.  This means that this changes introduces a change in behaviour because things happen in a slightly different order, but I don't see that it introduces any specific problems.

A separate observation here is that we're calling `isDisabled` separately for each command here (also in `schedule`).  If I'm reading the code correctly, this has two consequences:
* We're locking and unlocking a mutex for each command, which will make things slower and introduces the possibility of unnecessary lock contention.
* We're introducing the possibility of getting different answers every time we ask.  I'm not sure if there's something else that stops the `DriverStation` control word from being changed during `CommandScheduler.run`.  Imagine that the robot is disabled part way through `run` or disabled briefly and then re-enabled.  We might end up cancelling half the commands, which could be a bad state to leave the robot in.  It's probably better for a single invocation of `run` to have a consistent idea of robot state.

Reply:
> We're locking and unlocking a mutex for each command, which will make things slower and introduces the possibility of unnecessary lock contention.

What mutex is being referred to here?

Reply:
> What mutex is being referred to here?

The one in `DriverStation.isEnabled`.

Reply:
I'll just revert it back to having its own impl and give C++ its own impl as well.

Reply:
I wasn't objecting to the change in behaviour, just noting it.  There's a good argument to be made that we don't want to change the set of scheduled commands while we're doing our main `run` iteration on scheduled commands.